tests/cupy_tests/math_tests/test_sumprod.py

```diff
@@ -178,6 +178,128 @@ def test_prod_dtype(self, xp, src_dtype, dst_dtype):
         return a.prod(dtype=dst_dtype)
 
 
+@testing.parameterize(
+    *testing.product({
+        'shape': [(2, 3, 4), (20, 30, 40)],
+        'axis': [0, 1],
+        'transpose_axes': [True, False],
+        'keepdims': [True, False],
+        'func': ['nansum', 'nanprod']
+    })
+)
+@testing.gpu
+class TestNansumNanprodLong(unittest.TestCase):
+
+    def _do_transposed_axis_test(self):
+        return not self.transpose_axes and self.axis != 1
+
+    def _test(self, xp, dtype):
+        a = testing.shaped_arange(self.shape, xp, dtype)
+        if self.transpose_axes:
+            a = a.transpose(2, 0, 1)
+        if not issubclass(dtype, xp.integer):
+            a[:, 1] = xp.nan
+        func = getattr(xp, self.func)
+        return func(a, axis=self.axis, keepdims=self.keepdims)
+
+    @testing.for_float_dtypes()
+    @testing.for_int_dtypes()
```

Comment:
Nesting `for_*_dtypes()` decorators is not allowed.
Why not using `@testing.for_all_dtypes(no_bool=True)`?

Reply:
Sorry for the delay, and that's for telling me about nesting not working. I actually discovered some issues with complex and float16 types that I didn't see before, probably because the tests weren't running. :)

Reply:
I think everything's addressed now, would you mind taking a look at the changes?